var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -113,11 +113,14 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-cpp=cpp',
-                '--with-cxxcpp=cpp',
                 '--with-mpi=1',
                 '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
             ]
+        if sys.platform != "darwin":
```

Comment:
@jppelteret i suppose you are on Sierra? 

@eschnett would you mind adding `(spec.architecture.platform_os.version == "10.12")` to your `if`? 

Given extra feedback, i agree that this should be fixed to make things work.